src/manager.rs

```diff
@@ -76,8 +87,15 @@ impl<S: MutEventSubscriber> EventManager<S> {
     ///
     /// # Arguments
     ///
-    /// * `ready_events_capacity` : maximum number of ready events to be process in a single poll
+    /// * `ready_events_capacity`: maximum number of ready events to be
+    ///                            processed a single `run`. The maximum value of this
+    ///                            parameter is `EventManager::MAX_READY_EVENTS_CAPACITY`.
+    ///
     pub fn new_with_capacity(ready_events_capacity: usize) -> Result<Self> {
+        if ready_events_capacity as i32 > MAX_READY_EVENTS_CAPACITY {
```

Comment:
Is the conversion from `usize` to `i32` safe here?

Just noticed now that if we call `EventManager::<Arc<dyn EventSubscriber>>::new_with_capacity(usize::MAX)`, we panic [here](https://github.com/rust-vmm/event-manager/blob/master/src/epoll.rs#L31).

I think, it should be safer to convert MAX_READY_EVENTS_CAPACITY to `usize` (since MAX_READY_EVENTS_CAPACITY is a positive 32 bits number and usize is either u32 or u64):

This way we will return `InvalidCapacity` for usize::MAX too for example.

Reply:
That's makes sense. Good catch, I'll update it.